### PR TITLE
LPS-154286 Update xml description in workflow

### PR DESCRIPTION
**Pull request by M-IgorFranca**
https://issues.liferay.com/browse/LPS-154286

### CODE REVIEW

**Comment by M-IgorFranca** [comment]
ci:test:sf

**Comment by M-IgorFranca** [comment]
ci:test:sf

**Comment by M-IgorFranca** [comment]
ci:test:relevant

**Comment by M-IgorFranca** [comment]
ci:test:sf

**Comment by M-IgorFranca** [comment]
ci:reevaluate:1307401_3426

**Comment by lucasperj** [comment]
ci:test:relevant

**Comment by naraandrade** [comment]
Just started reviewing :)

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js [hunk 1]
```diff
@@ -89,11 +90,18 @@ export default function UpperToolbar({displayNames, languageIds}) {
 			xmlContent = currentEditor.getData();
 		}
 		else {
+			const deserializeUtil = new DeserializeUtil();
+			const xmlDefinition = currentEditor.getData();
```
Comment (naraandrade) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js:94]: @M-IgorFranca when we try to save/update the definition in the diagram view, the currentEditor is undefined. You need to check if the currentEditor is initialized to execute the code you added.

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js [hunk 2]
```diff
@@ -89,11 +90,18 @@ export default function UpperToolbar({displayNames, languageIds}) {
...
+			deserializeUtil.updateXMLDefinition(xmlDefinition);
```
Comment (naraandrade) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js:95]: @M-IgorFranca Add a blank line between the creation of the constant/variable and its usage.

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js [hunk 3]
```diff
@@ -89,11 +90,18 @@ export default function UpperToolbar({displayNames, languageIds}) {
...
+			const metadata = deserializeUtil.getMetadata();
+			setDefinitionDescription(metadata.description);
```
Comment (naraandrade) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js:97]: @M-IgorFranca Add a blank line between the creation of the constant/variable and its usage.

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js (hunk 1 above)
Comment (naraandrade) [reply modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js:94]: Also, this code only fix the problem with updating the description in the source view. If I add a node directly in the source view and try to save/update it, it is still not working.

**Review by naraandrade — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js (hunk 1 above)
Comment (M-IgorFranca) [reply modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js:94]: Sorry, i have one new updated version. You can check my modified? Thank you.

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js (hunk 2 above)
Comment (M-IgorFranca) [reply modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js:95]: ok

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js (hunk 1 above)
Comment (M-IgorFranca) [reply modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js]: Now I'm having problems with my afcb source format, I'm trying to fix it. In current version I can modify the xml description and solve the undefined problem. I don't understand why i can't pass in poshi tests in localhost.

**Comment by naraandrade** [comment]
Just started reviewing :)

**Review by naraandrade — CHANGES_REQUESTED** [review]
Hey @M-IgorFranca, good job on fixing the description update :) But we still have a problem with updating the whole definition when we are in the source view. The ticket only mentioned the description as an example, so I understand that you focused on that, but if we add changes in other parts of the code, it is not being saved/updated while we are in the source view. I will add that to the ticket.

**Comment by naraandrade** [comment]
Just started reviewing :)

**Comment by naraandrade** [comment]
ci:test:sf

**Comment by naraandrade** [comment]
Just started reviewing :)

**Review by naraandrade — APPROVED** [review]

**Comment by naraandrade** [comment]
ci:test:relevant

**Comment by naraandrade** [comment]
ci:forward

**Pull request closed without merging**